### PR TITLE
Fix DDP unused param error when TE is enabled in NeMo Lite

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
# What does this PR do ?

Fixed DDP issue with TE

### CODE REVIEW

File: examples/llm/sft/hf.py [hunk 1]
```diff
@@ -75,17 +75,9 @@ def squad(tokenizer) -> pl.LightningDataModule:
         grad_clip = None
     use_dist_samp = False
 
-    model = llm.HfAutoModelForCausalLM(args.model)
+    model = llm.HfAutoModelForCausalLM(model_name=args.model, model_accelerator=args.model_accelerator)
```
Comment (akoumpa) [inline examples/llm/sft/hf.py:78]: can we pass a partial function instead?

Comment (oyilmaz-nvidia) [reply examples/llm/sft/hf.py:78]: I also thought about it but I am just not sure how it'll look from the user perspective.

File: nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py [hunk 2]
```diff
@@ -78,6 +80,13 @@ def configure_model(self):
 
             config = AutoConfig.from_pretrained(self.model_name, trust_remote_code=self.trust_remote_code)
             self.model = AutoModelForCausalLM.from_config(config, trust_remote_code=self.trust_remote_code)
+
+        if self.model_accelerator is not None:
+            if self.model_accelerator == "te":
+                from nemo.lightning.pytorch.accelerate.transformer_engine import te_accelerate
+
+                te_accelerate(self.model, fp8_autocast=False)
```
[the remark this answers is not in the document]
Comment (oyilmaz-nvidia) [reply nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py:88]: Good question. I guess I should change that to a parameter class.

[the remark this answers is not in the document]
Comment (oyilmaz-nvidia) [reply nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py]: Ok, change it to partial function as @akoumpa suggested.

**Review by akoumpa — APPROVED** [review]

**Pull request merged**